improver/categorical/utilities.py

```diff
@@ -535,29 +537,46 @@ def check_tree(
                         f"Node {node} results in a bare category "
                         f"of {value} for the {result} condition. Should point to a leaf."
                     )
+            if "deterministic" in items and items["deterministic"] is True:
+                threshold_name = "thresholds"
+            else:
+                # Check diagnostic_conditions are all above or below.
+                diagnostic = items["diagnostic_conditions"]
+                tests_diagnostic = diagnostic
+                if isinstance(diagnostic[0], list):
+                    tests_diagnostic = [
+                        item for sublist in diagnostic for item in sublist
+                    ]
+                for value in tests_diagnostic:
+                    if value not in DIAGNOSTIC_CONDITIONS:
+                        issues.append(
+                            f"Node {node} uses invalid diagnostic condition "
+                            f"'{value}'; this should be 'above' or 'below'"
+                        )
 
-            # Check diagnostic_fields, diagnostic_conditions, and diagnostic_thresholds
-            # are all nested equivalently
-            if not _check_diagnostic_lists_consistency(items):
-                issues.append(
-                    f"Node {node} has inconsistent nesting for the "
-                    "diagnostic_fields, diagnostic_conditions, and "
-                    "diagnostic_thresholds fields"
-                )
+                # Check diagnostic_fields, diagnostic_conditions, and diagnostic_thresholds
+                # are all nested equivalently
+                if not _check_diagnostic_lists_consistency(items):
+                    issues.append(
+                        f"Node {node} has inconsistent nesting for the "
+                        "diagnostic_fields, diagnostic_conditions, and "
+                        "diagnostic_thresholds fields"
+                    )
+                threshold_name = "probability_thresholds"
 
-            # Check probability thresholds are numeric and there are as many of them
+            # Check probability thresholds/thresholds are numeric and there are as many of them
```

Comment:
I think I prefer this:
```suggestion
            # Check thresholds are numeric and there are as many of them
```

Reply:
Done